lib/timeutils/wallclocktime.h

```diff
@@ -124,6 +124,33 @@ struct _WallClockTime
 
 #endif
 
+static inline
+void dump_wall_clock_time(const WallClockTime *wct, GString *output)
+{
+  if (!wct || !output)
+    return;
+
+  g_string_append_printf(output, "WallClockTime:\n");
+  g_string_append_printf(output, "  Year: %d\n", wct->wct_year + 1900); // tm_year is years since 1900
+  g_string_append_printf(output, "  Month: %d\n", wct->wct_mon + 1);    // tm_mon is zero-based
+  g_string_append_printf(output, "  Day of Month: %d\n", wct->wct_mday);
+  g_string_append_printf(output, "  Weekday: %d\n", wct->wct_wday);
+  g_string_append_printf(output, "  Day of Year: %d\n", wct->wct_yday);
+  g_string_append_printf(output, "  Hour: %d\n", wct->wct_hour);
+  g_string_append_printf(output, "  Minute: %d\n", wct->wct_min);
+  g_string_append_printf(output, "  Second: %d\n", wct->wct_sec);
+  g_string_append_printf(output, "  Microsecond: %d\n", wct->wct_usec);
+  g_string_append_printf(output, "  DST: %d\n", wct->wct_isdst);
+
+#ifdef SYSLOG_NG_HAVE_STRUCT_TM_TM_GMTOFF
+  g_string_append_printf(output, "  GMT Offset: %ld\n", wct->wct_gmtoff);
+  g_string_append_printf(output, "  Timezone: %s\n", wct->wct_zone ? wct->wct_zone : "(null)");
+#else
+  g_string_append_printf(output, "  GMT Offset: %ld\n", wct->wct_gmtoff);
+  g_string_append_printf(output, "  Timezone: %s\n", wct->wct_zone ? wct->wct_zone : "(null)");
+#endif
```

Comment:
I don't see the difference in these two cases. The idea was to always have wct_gmtoff and wct_zone in WallClockTime so that we have those on all platforms. I don't think we need the conditional compilation.